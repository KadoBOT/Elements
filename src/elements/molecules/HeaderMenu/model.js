export type HeaderMenu = [string]

export type State = HeaderMenu[]
