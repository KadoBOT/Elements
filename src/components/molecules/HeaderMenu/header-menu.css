.header-menu {
  display: flex;
}

.header-menu__item {
  padding: 0px 30px;
  height: 50px;
}
