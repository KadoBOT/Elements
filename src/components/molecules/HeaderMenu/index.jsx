import React, {PropTypes} from 'react'

const HeaderMenu = () => {
  return(
    <div>
      Header Menu!
    </div>
  )
}

HeaderMenu.propTypes = {
  props: PropTypes.type
}

export default HeaderMenu
