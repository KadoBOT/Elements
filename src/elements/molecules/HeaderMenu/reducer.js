import { createReducer } from 'redux-act'
import type { State } from './model'

const initialState: State = ['Home', 'Careers', 'Blog', 'Contact']

export default createReducer({}, initialState)
