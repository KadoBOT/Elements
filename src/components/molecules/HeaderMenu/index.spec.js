import React from 'react'
import { shallow, render, mount } from 'enzyme';

import HeaderMenu from './'

it('should render the HeaderMenu', () => {
    const wrapper = shallow(
        <HeaderMenu menuItems={[]} />
    );
    expect(wrapper).toMatchSnapshot();
});
