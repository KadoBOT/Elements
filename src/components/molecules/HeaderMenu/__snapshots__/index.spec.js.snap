exports[`test should render the HeaderMenu 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="header-menu">
    
</div>,
  "nodes": Array [
    <div
      className="header-menu">
      
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <HeaderMenu
        menuItems={Array []} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "menuItems": Array [],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="header-menu">
          
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="header-menu">
          
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <HeaderMenu
    menuItems={Array []} />,
}
`;
